package.json

```diff
@@ -12,8 +12,8 @@
     "start": "NODE_ENV=production node build/server.js",
     "speculate": "speculate",
     "storybook": "start-storybook -p 9001 -c .storybook",
-    "test:e2e": "npx cypress run",
-    "test:e2e:interactive": "npx cypress open"
+    "test:e2e": "npm run dev & npx cypress run && pkill -f razzle",
```

Comment:
This script won't wait for the dev server to be running, and will run the Cypress test suite concurrently with the webpack build. It will only have worked by coincidence, because the `npx cypress run` command takes longer and than `npm run dev`.

Furthermore, what will happen if there is already a dev server running? I think the `npm run dev` command will fail, we should check that razzle is listening on that port before running the command, and either kill it and run `npm run dev` again, or just use that listening application.

I suggest that these scripts (this and the below) run a bash script file to do further checking of the above points.

Finally, I suggest we should consider running these commands against a production, not dev, build by default - with the option to test against a dev build.

Reply:
'the tests run against a production, not dev, build' - yes, makes sense to be default.

Reply:
@jtart Thanks for the review. 

I've read that adding a `wait` works for this situation - however, we'd need to supply a number of seconds, which is not suitable since start up time could be variable.  I will look for other alternatives. 

I was under the impression that the kill commands would be needed both locally and on CI. Why do you not want a tidy up task to run on CI? 

If we were to have different setup for the different environments, we could either have two different npm scripts - for local tests and CI tests or have conditional statements within a single script.  Do you have a particular preference? I personally prefer to separate things out rather than having conditionals.

Reply:
The commands are not applicable on a CI environment as CI will not start an application, and will only run tests against a running environment.

My preference for implementation would be a single script that only runs build/kill commands depending on the value of a `CI` environment variable, which already exists.

Reply:
Thanks!